openproblems/data/tabula_muris_senis.py

```diff
@@ -1,62 +1,47 @@
 from . import utils
 
 import os
-import pandas as pd
+import requests
 import scanpy as sc
 import scprep
 import tempfile
 
+COLLECTION_ID = "0b9d8a04-bb9d-44da-aa27-705bb65b54eb"
+DOMAIN = "cellxgene.cziscience.com"
+API_BASE = f"https://api.{DOMAIN}"
+METHOD_ALIASES = {"10x 3' v2": "droplet", "Smart-seq2": "facs"}
 
-def get_filenames_and_urls(url_df, method_list=None, organ_list=None):
-    """Takes in dataframe and returns corresponding filename(s) and url(s).
 
-    Takes in dataframe (with sample information stored), a list of methods,
-    and a list of organs.
-    Returns filenames and figshare URLs associated with inputs.
-    If method_list or organ_list are None, do not filter based on that argument.
-    """
-    subset_df = url_df.copy()
-    # If method_list specified, filter based on methods in list.
-    if method_list:
-        subset_df = subset_df.loc[subset_df.method.isin(method_list)]
-    # If organ_list specified, filter based on organs in list.
-    if organ_list:
-        subset_df = subset_df.loc[subset_df.organ.isin(organ_list)]
+def matching_dataset(dataset, method_list, organ_list):
+    # we want to skip combined datasets
+    if len(dataset["tissue"]) > 1:
+        return False
+    if dataset["tissue"][0]["label"] not in organ_list:
+        return False
```

Comment:
Do we want to enable full dataset download when we don't specify an organ?

Reply:
+1. If `organ_list` is None, then assume all organs.

Reply:
Done